h/static/scripts/controllers/autosuggest-dropdown-controller.js

```
@@ -0,0 +1,454 @@
+      this.options.classNames.activeItem = configOptions.classNames.activeItem || 'autosuggest__list-item--active';
+      this.options.classNames.header = configOptions.classNames.header || 'autosuggest__header';
+    }
+
+    // renaming simply to make it more obvious what
+    // the element is in other contexts the control
+    this._input = this.element;
+
+    // initial state values
+    this.setState({
+      visible: false,
+      header: configOptions.header || '',
+
+      // working list that are displayed to use
+      list: [],
+
+      // rootList is the original set that the reducer
+      // will receive to determine what should be shown
+      rootList: [],
+    });
+
+    this._setList(configOptions.list);
+  }
+
+  update(newState, prevState){
+
+    // if our prev state is empty then
+    // we assume that this is the first update/render call
+    if(!('visible' in prevState)){
+
+      // create the elements that make up the component
+      this._renderContentContainers();
+      this._addTopLevelEventListeners();
+    }
+
+    if(newState.visible !== prevState.visible){
+      // updates the dom to change the class which actually updates visibilty
+      setElementState(this._suggestionContainer, { open: newState.visible});
+    }
+
+    if(newState.header !== prevState.header){
+      this._header.innerHTML = newState.header;
+    }
+
+    if(this._listDidChange(newState.list, prevState.list)){
+      this._renderListItems();
+    }
+
+  }
+
+  /**
+   * _setHeader - sets what would be the top bolded header text
+   *  to give context of what the suggestions are for
+   *
+   * @param  {string} header Text to place in header
+   */
+  _setHeader(header){
+    this.setState({
+      header,
+    });
+  }
+
+  /**
+   * _setList - update the current list
+   *
+   * @param  {Array} list The new list.
+   */
+  _setList(list){
+
+    if(!Array.isArray(list)){
+      console.error('setList requires an array first argument');
```

I would suggest throwing a `TypeError` rather than just logging an error here. That way the error will get reported back to us via Raven.